src/Cms/ModelWithContent.php

```diff
@@ -375,11 +375,17 @@ public function query(string $query = null, string $expect = null)
 	 */
 	public function readContent(string $languageCode = null): array
 	{
-		try {
-			return Data::read($this->contentFile($languageCode));
-		} catch (Throwable) {
+		$file = $this->contentFile($languageCode);
+
+		// only if the content file really does not exist, it's ok
+		// to return empty content. Otherwise this could lead to
+		// content loss in case of file reading exceptions
+		// (i.e. if the file is locked)
+		if (file_exists($file) === false) {
 			return [];
 		}
+
+		return Data::read($file);
```

Comment:
Ok, that's true and another potential problem, but not the cause of the UUID issue. I don't think we should fix more theoretical issues at this point. The problems that we have with this are already big enough on their own :)

Reply:
The fix for this is pretty simple. I'll push a fix over the weekend.

Reply:
Given how we set up `Handler::read()` and `F::read()`, it's not as simple as expected because there can be both PHP warnings and exceptions to take care of.

So I think we can ship it like this for now. It may generate more false-negative error messages, but at least there won't be undetected read errors.